Makefile

```diff
@@ -155,21 +159,21 @@ fmt: format.gofmt format.goimports # backward compatible with ./bin/fmt.sh
 check: check.vet check.lint
 
 format.gofmt: ; $(info $(H) formatting files with go fmt...)
-	$(Q) gofmt -s -w $(GO_FILES)
+	$(Q) gofmt -s -w $$($(GO_FILES_CMD))
```

Comment:
what does that execute ?
```
gofmt -s -w $(find . -name '*.go' | grep -v -E '$(GO_EXCLUDE)') 
```

it does mean if you run all the targets you keep running the find instead of running it once no ?

Reply:
I'm not sure I understand the question.  Prior to this change 'make' sees
<pre>
GO_FILES := $(shell find . -name '*.go' | grep -v -E '$(GO_EXCLUDE)')
</pre>

and tries to run find & grep each time it parses the Makefile.  After the change the intent is to just have:

<pre>
GO_FILES_CMD := find . -name '*.go' | grep -v -E '$(GO_EXCLUDE)'
</pre>
which make expands to the string:
<pre>
GO_FILES_CMD := find . -name '*.go' | grep -v -E '/vendor/|.pb.go|.gen.go'
</pre>
when a "make format.gofmt" is run make thinks its running:
<pre>
gofmt -s -w $(find . -name '*.go' | grep -v -E '/vendor/|.pb.go|.gen.go')
</pre>
the shell then runs what's within the parenthesis and provides the resulting list as parameters to "gofmt -s -w".  Does this address the question or is there something I might be overlooking (or got wrong in my thinking)?

Reply:
I meant that GO_FILES_CMD is used by more than 1 target, I didn't examine the dependencies or if all those targets end up indirectly invoked together or not

but if the most common case is people don't need that file list (and get the 1s saving), sgtm

Reply:
Ah, you're probably right that if someone does a "make fmt" then this will try to "build" format.gofmt and  format.goimports , so that use case takes a 1 second hit while everyone else should see a 1 second win.